### ISSUE TITLE
paragraph-level translations and multilingual tts

### ISSUE THREAD

**Issue by elias-sundqvist** [nanihadesuka/NovelDokusha#98]
This is actually multiple feature requests, but they are related. 

I have some キノの旅 epubs that I would like to read. I'm not very good at Japanese, but it would be an opportunity to learn. 

Ideally, instead of translating the entire text into a new language (and replacing the original content), I would like to retain the original text, and see the translation below each paragraph.  I would also like to be able to listen to this like an audiobook, where I have a Japanese TTS voice for the Japanese text and an English TTS voice for the translated text. 

Also, with the current epub import, my entire epub is interpreted as one long chapter, so it would be good if the translations could be loaded dynamically instead of processing everything in the beginning. Because that takes forever.  

tldr;
1. Translate one paragraph at a time, on demand instead of everything at once
2. Show translation below each paragraph, instead of replacing
3. Use one TTS voice for the original text and another for the translated text

I would be willing to help out with this, but I could use some assistance with parts, since I'm not used to developing UI things for android. 

What are your thoughts about this @nanihadesuka? Does it sound reasonable, or is it out-of-scope for your goals with this app?

**Comment by elias-sundqvist** [comment]
It could perhaps look something like this 
The translation (yellow) is only visible when the paragraph is focused.
![image](https://github.com/nanihadesuka/NovelDokusha/assets/9102856/b62efde4-37f8-44f9-b294-2099f7c7cf17)

**Comment by nanihadesuka** [comment]
The idea looks good but the feature is complex to add as it interacts with all the other parts of the reader features.

I was planning on moving the reader UI to jetpack compose to deal with the ui side more easily (will help implement this new feature more easily too)

I'm willing to implement what you ask as is useful  but I will probably start in 2 months from now. 

If you want you can try to see how it works and try to add it.

**Comment by nanihadesuka** [comment]
Will not all in the foreseeable future.
